packages/protocol/contracts/stability/SortedOracles.sol

```diff
@@ -273,6 +290,19 @@ contract SortedOracles is ISortedOracles, ICeloVersionedContract, Ownable, Initi
     return oracles[token];
   }
 
+  /**
+   * @notice Returns the expiry for the token if exists, if not the default
+   * @param token The address of the token
+   * @return The report expiry in seconds
+   */
+  function getTokenReportExpirySeconds(address token) public view returns (uint256) {
+    if (tokenReportExpirySeconds[token] == 0) {
```

Comment:
Is it safe to always fallback on the old default?

Reply:
This is a good question. In the way I see it, no token should ever be relying on this value as it feels "unsafe" and not precisely accurate.

I don't know what are the backwards compatibility/upgradeability considerations that we need to have here, but I imagine that every token which the contract is receiving reports for should have a registered report expiry duration.

Reply:
The way I see it there's a contract level value which is *always* set because it's part of the constructor, so even if that's `0` it was set intentionally as `0`.

Whereas there's no clear point where one would register a new token that's being handled by an instance of the contract. Therefore there's no way to enforce that a token has a specific expiry set.

So the way I look at it is, we have a global value that we know is always set at contract creation and then we have optional per token values that may or may not be set. This feels safer to me rather than relying on having a mapping for a token created before we start reporting for that token. That's a state that's unenforceable without major changes.

Reply:
> This feels safer to me rather than relying on having a mapping for a token created before we start reporting for that token. 

What I had in mind is that both the token and the reportExpirySeconds mapping should always be added together in a single governance proposal (and that in the upgrade process for this particular contract we should add the initial mapping for cUSD) which is why to me using a fallback value feels more like a mistake on us forgetting to add the mapping than an intentional side effect.

Reply:
I don't really like the fact that we're encoding this dependency: "reportExpirySeconds mapping should always be added together in a single governance proposal" in our heads, but there's no way to encode that explicitly in the contract.

Reply:
Had a discussion with @bowd about this. I originally assumed that there was a notion of "adding" a token to the contract before being able to report to it, and hence why I imagined that we would always add a token together with its reportExpirySeconds value. This is not the case, as the only thing needed before being able to report values to a token is to register Oracles to it.

Given that there's no way to enforce that a token must have a reportExpirySeconds I am fine with with keeping things as they are. We also discussed that one could imagine the reportExpirySeconds value being the default for stable tokens native to the protocol (cUSD, cEUR, etc) while the special mapping is only used for special tokens that might have some different requirements.

TL;DR: Fine with keeping things as they are unless you have any big concerns @m-chrzan

Reply:
Isn't it possible to extend the contract to check when an oracle reports that there is a token-specific expiry?

Reply:
That sounds good to me, thanks for the explanations @bowd and @nvtaveras.